enterprise/frontend/src/embedding/data-picker/DataSelector.jsx

```diff
@@ -465,50 +400,17 @@ export class UnconnectedDataSelector extends Component {
     return this.hasModels() && this.props.hasNestedQueriesEnabled;
   };
 
-  hasMetrics = () => {
-    const { availableModels, canSelectMetric, loaded } = this.props;
-    return loaded && canSelectMetric && availableModels.includes("metric");
-  };
-
-  hasUsableMetrics = () => {
-    // As metrics are actually saved questions, nested queries must be enabled
-    return this.hasMetrics() && this.props.hasNestedQueriesEnabled;
-  };
-
-  hasUsableModelsOrMetrics = () => {
-    return this.hasUsableModels() || this.hasUsableMetrics();
-  };
-
-  hasSavedQuestions = () => {
-    const { canSelectSavedQuestion } = this.props;
-    return (
-      this.state.databases.some((database) => database.is_saved_questions) &&
-      canSelectSavedQuestion
-    );
-  };
-
   getDatabases = () => {
     const { databases } = this.state;
 
-    // When there is at least one dataset,
-    // "Saved Questions" are presented in a different picker step
-    // So it should be excluded from a regular databases list
-    const shouldRemoveSavedQuestionDatabaseFromList =
-      !this.props.hasNestedQueriesEnabled ||
-      this.hasUsableModelsOrMetrics() ||
-      !this.props.canSelectSavedQuestion;
```

Comment:
This was added in https://github.com/metabase/metabase/pull/54845/files#r1989033787. And for embedding use case, we set it to `false`

https://github.com/metabase/metabase/blob/319b45efb0028e718bbfa9fed02771c03b994f00/frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx#L112

That means this expression always translates to `true`, and `shouldRemoveSavedQuestionDatabaseFromList` is always `true`, so we could skip this check entirely.